File: codes/go/Array.some/Array.some.go
package main

import (
	"encoding/json"
	"fmt"
	"reflect"
)

const EMPTY_STRING = ""
const TAB = "    "

type Any interface{}
type Object map[string]Any

func prettyJsonStringify(anything Any) string {
	marshalledJson, err := json.MarshalIndent(anything, EMPTY_STRING, TAB)
	if err == nil {
		return string(marshalledJson)
	}

	return "undefined"
}

func prettyArrayOfPrimitives(anArray []Any) string {
	result := "["
	for arrayItemIndex, arrayItem := range anArray {
		arrayItemType := reflect.TypeOf(arrayItem).Kind()
		if arrayItemType != reflect.Float32 &&
			arrayItemType != reflect.Float64 &&
			arrayItemType != reflect.Int &&
			arrayItemType != reflect.Int8 &&
			arrayItemType != reflect.Int16 &&
			arrayItemType != reflect.Int32 &&
			arrayItemType != reflect.Int64 &&
			arrayItemType != reflect.Uint &&
			arrayItemType != reflect.Uint8 &&
			arrayItemType != reflect.Uint16 &&
			arrayItemType != reflect.Uint32 &&
			arrayItemType != reflect.Uint64 &&
			arrayItemType != reflect.String {
			continue
		}
		if arrayItemType == reflect.String {
			stringArrayItem := arrayItem.(string)
			result = result + "\"" + stringArrayItem + "\""
		}
		if arrayItemType == reflect.Float32 {
			stringArrayItem := fmt.Sprint(arrayItem.(float32))
			result = result + stringArrayItem
		}
		if arrayItemType == reflect.Float64 {
			stringArrayItem := fmt.Sprint(arrayItem.(float64))
			result = result + stringArrayItem
		}
		if arrayItemType == reflect.Int {
			stringArrayItem := fmt.Sprint(arrayItem.(int))
			result = result + stringArrayItem
		}
		if arrayItemType == reflect.Int8 {
			stringArrayItem := fmt.Sprint(arrayItem.(int8))
			result = result + stringArrayItem
		}
		if arrayItemType == reflect.Int16 {
			stringArrayItem := fmt.Sprint(arrayItem.(int16))
			result = result + stringArrayItem
		}
		if arrayItemType == reflect.Int32 {
			stringArrayItem := fmt.Sprint(arrayItem.(int32))
			result = result + stringArrayItem
		}
		if arrayItemType == reflect.Int64 {
			stringArrayItem := fmt.Sprint(arrayItem.(int64))
			result = result + stringArrayItem
		}
		if arrayItemType == reflect.Uint {
			stringArrayItem := fmt.Sprint(arrayItem.(uint))
			result = result + stringArrayItem
		}
		if arrayItemType == reflect.Uint8 {
			stringArrayItem := fmt.Sprint(arrayItem.(uint8))
			result = result + stringArrayItem
		}
		if arrayItemType == reflect.Uint16 {
			stringArrayItem := fmt.Sprint(arrayItem.(uint16))
			result = result + stringArrayItem
		}
		if arrayItemType == reflect.Uint32 {
			stringArrayItem := fmt.Sprint(arrayItem.(uint32))
			result = result + stringArrayItem
		}
		if arrayItemType == reflect.Int64 {
			stringArrayItem := fmt.Sprint(arrayItem.(uint64))
			result = result + stringArrayItem
		}
		if (arrayItemIndex + 1) != len(anArray) {
			result = result + ", "
		}
	}
	result = result + "]"
	return result
}

func arraySomeV1(callbackFunction func(Any, int, []Any) bool, anArray []Any) bool {
	// JavaScript-like Array.some() function
	isConditionMatch := false
	for arrayItemIndex, arrayItem := range anArray {
		isConditionMatch = callbackFunction(arrayItem, arrayItemIndex, anArray)
		if isConditionMatch == true {
			break
		}
	}
	return isConditionMatch
}

func arraySomeV2(callbackFunction func(Any, int, []Any) bool, anArray []Any) bool {
	// JavaScript-like Array.some() function
	isConditionMatch := false
	for arrayItemIndex, arrayItem := range anArray {
		isConditionMatch = callbackFunction(arrayItem, arrayItemIndex, anArray)
		if isConditionMatch == true {
			return isConditionMatch
		}
	}
	return isConditionMatch
}

func arraySomeV3(callbackFunction func(Any, int, []Any) bool, anArray []Any) bool {
	// JavaScript-like Array.some() function
	for arrayItemIndex, arrayItem := range anArray {
		isConditionMatch := callbackFunction(arrayItem, arrayItemIndex, anArray)
		if isConditionMatch == true {
			return true
		}
	}
	return false
}

func arraySomeV4(callbackFunction func(Any, int, []Any) bool, anArray []Any) bool {
	// JavaScript-like Array.some() function
	for arrayItemIndex, arrayItem := range anArray {
		if callbackFunction(arrayItem, arrayItemIndex, anArray) == true {
			return true
		}
	}
	return false
}

func main() {
	fmt.Println("\n// JavaScript-like Array.some() in Go Slice")

	numbers := []Any{12, 34, 27, 23, 65, 93, 36, 87, 4, 254}
	fmt.Println("numbers:", prettyArrayOfPrimitives(numbers))

	var isAnyNumberLessThan500 bool
	var isAnyNumberMoreThan500 bool

	fmt.Println("// using JavaScript-like Array.some() function \"arraySomeV1\"")

	isAnyNumberLessThan500 = arraySomeV1(func(number Any, _ int, _ []Any) bool {
		return number.(int) < 500
	}, numbers)
	fmt.Println("is any number < 500: ", prettyJsonStringify(isAnyNumberLessThan500))
	// is any number < 500: true

	isAnyNumberMoreThan500 = arraySomeV1(func(number Any, _ int, _ []Any) bool {
		return number.(int) > 500
	}, numbers)
	fmt.Println("is any number > 500: ", prettyJsonStringify(isAnyNumberMoreThan500))
	// is any number > 500: false

	fmt.Println("// using JavaScript-like Array.some() function \"arraySomeV2\"")

	isAnyNumberLessThan500 = arraySomeV2(func(number Any, _ int, _ []Any) bool {
		return number.(int) < 500
	}, numbers)
	fmt.Println("is any number < 500: ", prettyJsonStringify(isAnyNumberLessThan500))
	// is any number < 500: true

	isAnyNumberMoreThan500 = arraySomeV2(func(number Any, _ int, _ []Any) bool {
		return number.(int) > 500
	}, numbers)
	fmt.Println("is any number > 500: ", prettyJsonStringify(isAnyNumberMoreThan500))
	// is any number > 500: false

	fmt.Println("// using JavaScript-like Array.some() function \"arraySomeV3\"")

	isAnyNumberLessThan500 = arraySomeV3(func(number Any, _ int, _ []Any) bool {
		return number.(int) < 500
	}, numbers)
	fmt.Println("is any number < 500: ", prettyJsonStringify(isAnyNumberLessThan500))
	// is any number < 500: true

	isAnyNumberMoreThan500 = arraySomeV3(func(number Any, _ int, _ []Any) bool {
		return number.(int) > 500
	}, numbers)
	fmt.Println("is any number > 500: ", prettyJsonStringify(isAnyNumberMoreThan500))
	// is any number > 500: false

	fmt.Println("// using JavaScript-like Array.some() function \"arraySomeV4\"")

	isAnyNumberLessThan500 = arraySomeV4(func(number Any, _ int, _ []Any) bool {
		return number.(int) < 500
	}, numbers)
	fmt.Println("is any number < 500: ", prettyJsonStringify(isAnyNumberLessThan500))
	// is any number < 500: true

	isAnyNumberMoreThan500 = arraySomeV4(func(number Any, _ int, _ []Any) bool {
		return number.(int) > 500
	}, numbers)
	fmt.Println("is any number > 500: ", prettyJsonStringify(isAnyNumberMoreThan500))
	// is any number > 500: false

	fmt.Println("\n// JavaScript-like Array.some() in Slice of Maps")

	products := []Any{
		Object{
			"code":  "pasta",
			"price": 321,
		},
		Object{
			"code":  "bubble_gum",
			"price": 233,
		},
		Object{
			"code":  "potato_chips",
			"price": 5,
		},
		Object{
			"code":  "towel",
			"price": 499,
		},
	}

	fmt.Println("products:" + prettyJsonStringify(products))

	var isAnyProductPriceLessThan500 bool
	var isAnyProductPriceMoreThan500 bool

	fmt.Println("// using JavaScript-like Array.some() function \"arraySomeV1\"")

	isAnyProductPriceLessThan500 = arraySomeV1(func(product Any, _ int, _ []Any) bool {
		return product.(Object)["price"].(int) < 500
	}, products)
	fmt.Println("is any product price < 500:", prettyJsonStringify(isAnyProductPriceLessThan500))
	// is any product price < 500: true

	isAnyProductPriceMoreThan500 = arraySomeV1(func(product Any, _ int, _ []Any) bool {
		return product.(Object)["price"].(int) > 500
	}, products)
	fmt.Println("is any product price > 500:", prettyJsonStringify(isAnyProductPriceMoreThan500))
	// is any product price > 500: false

	fmt.Println("// using JavaScript-like Array.some() function \"arraySomeV2\"")

	isAnyProductPriceLessThan500 = arraySomeV2(func(product Any, _ int, _ []Any) bool {
		return product.(Object)["price"].(int) < 500
	}, products)
	fmt.Println("is any product price < 500:", prettyJsonStringify(isAnyProductPriceLessThan500))
	// is any product price < 500: true

	isAnyProductPriceMoreThan500 = arraySomeV2(func(product Any, _ int, _ []Any) bool {
		return product.(Object)["price"].(int) > 500
	}, products)
	fmt.Println("is any product price > 500:", prettyJsonStringify(isAnyProductPriceMoreThan500))
	// is any product price > 500: false

	fmt.Println("// using JavaScript-like Array.some() function \"arraySomeV3\"")

	isAnyProductPriceLessThan500 = arraySomeV3(func(product Any, _ int, _ []Any) bool {
		return product.(Object)["price"].(int) < 500
	}, products)
	fmt.Println("is any product price < 500:", prettyJsonStringify(isAnyProductPriceLessThan500))
	// is any product price < 500: true

	isAnyProductPriceMoreThan500 = arraySomeV3(func(product Any, _ int, _ []Any) bool {
		return product.(Object)["price"].(int) > 500
	}, products)
	fmt.Println("is any product price > 500:", prettyJsonStringify(isAnyProductPriceMoreThan500))
	// is any product price > 500: false

	fmt.Println("// using JavaScript-like Array.some() function \"arraySomeV4\"")

	isAnyProductPriceLessThan500 = arraySomeV4(func(product Any, _ int, _ []Any) bool {
		return product.(Object)["price"].(int) < 500
	}, products)
	fmt.Println("is any product price < 500:", prettyJsonStringify(isAnyProductPriceLessThan500))
	// is any product price < 500: true

	isAnyProductPriceMoreThan500 = arraySomeV4(func(product Any, _ int, _ []Any) bool {
		return product.(Object)["price"].(int) > 500
	}, products)
	fmt.Println("is any product price > 500:", prettyJsonStringify(isAnyProductPriceMoreThan500))
	// is any product price > 500: false
}

File: codes/visual_basic_dotnet/array_some.vb
Imports System.Collections.Generic ' Dictionary, KeyValuePair

Module Program
    Function PrettyArrayOfPrimitives(ByVal AnArrayOfPrimitives As List(Of Object)) As String
        Dim Result As String = "["
        Dim ArrayItemIndex As Integer = 0
        For Each ArrayItem As Object In AnArrayOfPrimitives
            If (((TypeOf ArrayItem Is String) = false) AndAlso (IsNumeric(ArrayItem) = false) AndAlso ((TypeOf ArrayItem Is Boolean) = false) AndAlso ((ArrayItem Is Nothing) = false)) Then Continue For
            If (TypeOf ArrayItem Is String) Then Result += """" & ArrayItem & """"
            If (IsNumeric(ArrayItem) = true) Then Result += CStr(ArrayItem).Replace(",", ".")
            If (TypeOf ArrayItem Is Boolean) Then Result += CStr(ArrayItem)
            If (ArrayItem Is Nothing) Then Result += "null"
            If ((ArrayItemIndex + 1) <> AnArrayOfPrimitives.Count) Then Result += ", "
            ArrayItemIndex += 1
        Next
        Result += "]"
        Return Result
    End Function

    Function PrettyJsonStringify(ByVal Anything As Object, Optional ByVal Indent As String = "    ") As String
        Dim IndentLevel As Integer = 0
        Dim PrettyJsonStringifyInner As Func(Of Object, String, String) = Function(ByVal AnythingInner As Object, ByVal IndentInner As String)
            If (AnythingInner Is Nothing) Then Return "null"
            If (TypeOf AnythingInner Is String) Then Return """" & AnythingInner & """"
            If (IsNumeric(AnythingInner) = true) Then Return CStr(AnythingInner).Replace(",", ".")
            If (TypeOf AnythingInner Is Boolean) Then Return CStr(AnythingInner)
            If (TypeOf AnythingInner Is List(Of Object)) Then
                If (AnythingInner.Count = 0) Then Return "[]"
                IndentLevel += 1
                Dim Result As String = "[" & Environment.NewLine & String.Concat(Enumerable.Repeat(IndentInner, IndentLevel))
                Dim ArrayItemIndex As Integer = 0
                For Each ArrayItem As Object In AnythingInner
                    Result &= PrettyJsonStringifyInner(ArrayItem, IndentInner)
                    If ((ArrayItemIndex + 1) <> AnythingInner.Count) Then Result &= "," & Environment.NewLine & String.Concat(Enumerable.Repeat(IndentInner, IndentLevel))
                    ArrayItemIndex += 1
                Next
                IndentLevel -= 1
                Result &= Environment.NewLine & String.Concat(Enumerable.Repeat(IndentInner, IndentLevel)) & "]"
                Return Result
            End If
            If (TypeOf AnythingInner Is Dictionary(Of String, Object)) Then
                If (AnythingInner.Count = 0) Then Return "{}"
                IndentLevel += 1
                Dim Result As String = "{" & Environment.NewLine & String.Concat(Enumerable.Repeat(IndentInner, IndentLevel))
                Dim ObjectIterationIndex As Integer = 0
                For Each ObjectEntry As KeyValuePair(Of String, Object) In AnythingInner
                    Dim ObjectKey = ObjectEntry.Key
                    Dim ObjectValue = ObjectEntry.Value
                    Result &= """" & ObjectKey & """: " & PrettyJsonStringifyInner(ObjectValue, IndentInner)
                    If ((ObjectIterationIndex + 1) <> AnythingInner.Count) Then Result &= "," & Environment.NewLine & String.Concat(Enumerable.Repeat(IndentInner, IndentLevel))
                    ObjectIterationIndex += 1
                Next
                IndentLevel -= 1
                Result &= Environment.NewLine & String.Concat(Enumerable.Repeat(IndentInner, IndentLevel)) & "}"
                Return Result
            End If
            Return "null"
        End Function
        Return PrettyJsonStringifyInner(Anything, Indent)
    End Function

    Function ArraySomeV1(ByVal CallbackFunction As Func(Of Object, Integer, List(Of Object), Boolean), ByVal AnArray As List(Of Object)) As Boolean
        ' JavaScript-like Array.some() function
        Dim IsConditionMatch As Boolean = false
        Dim ArrayItemIndex As Integer = 0
        For Each ArrayItem As Object In AnArray
            IsConditionMatch = CallbackFunction(ArrayItem, ArrayItemIndex, AnArray)
            If (IsConditionMatch = true) Then Exit For
            ArrayItemIndex += 1
        Next
        Return IsConditionMatch
    End Function

    Function ArraySomeV2(ByVal CallbackFunction As Func(Of Object, Integer, List(Of Object), Boolean), ByVal AnArray As List(Of Object)) As Boolean
        ' JavaScript-like Array.some() function
        Dim IsConditionMatch As Boolean = false
        Dim ArrayItemIndex As Integer = 0
        For Each ArrayItem As Object In AnArray
            IsConditionMatch = CallbackFunction(ArrayItem, ArrayItemIndex, AnArray)
            If (IsConditionMatch = true) Then Return IsConditionMatch
            ArrayItemIndex += 1
        Next
        Return IsConditionMatch
    End Function

    Function ArraySomeV3(ByVal CallbackFunction As Func(Of Object, Integer, List(Of Object), Boolean), ByVal AnArray As List(Of Object)) As Boolean
        ' JavaScript-like Array.some() function
        Dim ArrayItemIndex As Integer = 0
        For Each ArrayItem As Object In AnArray
            Dim IsConditionMatch As Boolean = CallbackFunction(ArrayItem, ArrayItemIndex, AnArray)
            If (IsConditionMatch = true) Then Return true
            ArrayItemIndex += 1
        Next
        Return false
    End Function

    Function ArraySomeV4(ByVal CallbackFunction As Func(Of Object, Integer, List(Of Object), Boolean), ByVal AnArray As List(Of Object)) As Boolean
        ' JavaScript-like Array.some() function
        Dim ArrayItemIndex As Integer = 0
        For Each ArrayItem As Object In AnArray
            If (CallbackFunction(ArrayItem, ArrayItemIndex, AnArray) = true) Then Return true
            ArrayItemIndex += 1
        Next
        Return false
    End Function

    Sub Main(Args As String())
        Console.WriteLine($"{Environment.NewLine}' JavaScript-like Array.some() in Visual Basic (.NET) List")

        Dim Numbers As List(Of Object) = New List(Of Object) From {12, 34, 27, 23, 65, 93, 36, 87, 4, 254}
        Console.WriteLine($"Numbers: {PrettyArrayOfPrimitives(Numbers)}")

        Dim IsAnyNumberLessThan500 As Boolean
        Dim IsAnyNumberMoreThan500 As Boolean

        Console.WriteLine("' using JavaScript-like Array.some() function ""ArraySomeV1""")

        IsAnyNumberLessThan500 = ArraySomeV1(Function(ByVal Number As Integer, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Number < 500), Numbers)
        Console.WriteLine($"is any number < 500: {IsAnyNumberLessThan500}")
        ' is any number < 500: True

        IsAnyNumberMoreThan500 = ArraySomeV1(Function(ByVal Number As Integer, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Number > 500), Numbers)
        Console.WriteLine($"is any number > 500: {IsAnyNumberMoreThan500}")
        ' is any number > 500: False

        Console.WriteLine("' using JavaScript-like Array.some() function ""ArraySomeV2""")

        IsAnyNumberLessThan500 = ArraySomeV2(Function(ByVal Number As Integer, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Number < 500), Numbers)
        Console.WriteLine($"is any number < 500: {IsAnyNumberLessThan500}")
        ' is any number < 500: True

        IsAnyNumberMoreThan500 = ArraySomeV2(Function(ByVal Number As Integer, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Number > 500), Numbers)
        Console.WriteLine($"is any number > 500: {IsAnyNumberMoreThan500}")
        ' is any number > 500: False

        Console.WriteLine("' using JavaScript-like Array.some() function ""ArraySomeV3""")

        IsAnyNumberLessThan500 = ArraySomeV3(Function(ByVal Number As Integer, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Number < 500), Numbers)
        Console.WriteLine($"is any number < 500: {IsAnyNumberLessThan500}")
        ' is any number < 500: True

        IsAnyNumberMoreThan500 = ArraySomeV3(Function(ByVal Number As Integer, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Number > 500), Numbers)
        Console.WriteLine($"is any number > 500: {IsAnyNumberMoreThan500}")
        ' is any number > 500: False

        Console.WriteLine("' using JavaScript-like Array.some() function ""ArraySomeV4""")

        IsAnyNumberLessThan500 = ArraySomeV4(Function(ByVal Number As Integer, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Number < 500), Numbers)
        Console.WriteLine($"is any number < 500: {IsAnyNumberLessThan500}")
        ' is any number < 500: True

        IsAnyNumberMoreThan500 = ArraySomeV4(Function(ByVal Number As Integer, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Number > 500), Numbers)
        Console.WriteLine($"is any number > 500: {IsAnyNumberMoreThan500}")
        ' is any number > 500: False

        Console.WriteLine("' using Visual Basic (.NET) Array.some() built-in method ""Array.Any()""")

        IsAnyNumberLessThan500 = Numbers.Any(Function(ByVal Number As Integer) (Number < 500))
        Console.WriteLine($"is any number < 500: {IsAnyNumberLessThan500}")
        ' is any number < 500: True

        IsAnyNumberMoreThan500 = Numbers.Any(Function(ByVal Number As Integer) (Number > 500))
        Console.WriteLine($"is any number > 500: {IsAnyNumberMoreThan500}")
        ' is any number > 500: False

        Console.WriteLine($"{Environment.NewLine}' JavaScript-like Array.some() in Visual Basic (.NET) List of Dictionaries")

        Dim Products As List(Of Object) = New List(Of Object) From {
            New Dictionary(Of String, Object) From {
                {"code", "pasta"},
                {"price", 321}
            },
            New Dictionary(Of String, Object) From {
                {"code", "bubble_gum"},
                {"price", 233}
            },
            New Dictionary(Of String, Object) From {
                {"code", "potato_chips"},
                {"price", 5}
            },
            New Dictionary(Of String, Object) From {
                {"code", "towel"},
                {"price", 499}
            }
        }
        Console.WriteLine($"Products: {PrettyJsonStringify(Products)}")

        Dim IsAnyProductPriceLessThan500 As Boolean
        Dim IsAnyProductPriceMoreThan500 As Boolean

        Console.WriteLine("' using JavaScript-like Array.some() function ""ArraySomeV1""")

        IsAnyProductPriceLessThan500 = ArraySomeV1(Function(ByVal Product As Object, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Product("price") < 500), Products)
        Console.WriteLine($"is any product price < 500: {IsAnyProductPriceLessThan500}")
        ' is any product price < 500: True

        IsAnyProductPriceMoreThan500 = ArraySomeV1(Function(ByVal Product As Object, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Product("price") > 500), Products)
        Console.WriteLine($"is any product price > 500: {IsAnyProductPriceMoreThan500}")
        ' is any product price > 500: False

        Console.WriteLine("' using JavaScript-like Array.some() function ""ArraySomeV2""")

        IsAnyProductPriceLessThan500 = ArraySomeV2(Function(ByVal Product As Object, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Product("price") < 500), Products)
        Console.WriteLine($"is any product price < 500: {IsAnyProductPriceLessThan500}")
        ' is any product price < 500: True

        IsAnyProductPriceMoreThan500 = ArraySomeV2(Function(ByVal Product As Object, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Product("price") > 500), Products)
        Console.WriteLine($"is any product price > 500: {IsAnyProductPriceMoreThan500}")
        ' is any product price > 500: False

        Console.WriteLine("' using JavaScript-like Array.some() function ""ArraySomeV3""")

        IsAnyProductPriceLessThan500 = ArraySomeV3(Function(ByVal Product As Object, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Product("price") < 500), Products)
        Console.WriteLine($"is any product price < 500: {IsAnyProductPriceLessThan500}")
        ' is any product price < 500: True

        IsAnyProductPriceMoreThan500 = ArraySomeV3(Function(ByVal Product As Object, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Product("price") > 500), Products)
        Console.WriteLine($"is any product price > 500: {IsAnyProductPriceMoreThan500}")
        ' is any product price > 500: False

        Console.WriteLine("' using JavaScript-like Array.some() function ""ArraySomeV4""")

        IsAnyProductPriceLessThan500 = ArraySomeV4(Function(ByVal Product As Object, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Product("price") < 500), Products)
        Console.WriteLine($"is any product price < 500: {IsAnyProductPriceLessThan500}")
        ' is any product price < 500: True

        IsAnyProductPriceMoreThan500 = ArraySomeV4(Function(ByVal Product As Object, ByVal ArrayItemIndex As Integer, ByVal AnArray As List(Of Object)) (Product("price") > 500), Products)
        Console.WriteLine($"is any product price > 500: {IsAnyProductPriceMoreThan500}")
        ' is any product price > 500: False

        Console.WriteLine("' using Visual Basic (.NET) Array.some() built-in method ""Array.Any()""")

        IsAnyProductPriceLessThan500 = Products.Any(Function(ByVal Product As Object) (Product("price") < 500))
        Console.WriteLine($"is any product price < 500: {IsAnyProductPriceLessThan500}")
        ' is any product price < 500: True

        IsAnyProductPriceMoreThan500 = Products.Any(Function(ByVal Product As Object) (Product("price") > 500))
        Console.WriteLine($"is any product price > 500: {IsAnyProductPriceMoreThan500}")
        ' is any product price > 500: False
    End Sub
End Module
